### PR TITLE
Provide SigningService creation via owned String (#1114)

### PR DESCRIPTION
**Pull request by mchoicpe-amazon**
## Motivation and Context
`SigningService` has inner type `Cow<'static, str>` but does not allow creation with owned `String`. This causes limitations in its functionality to users.
https://github.com/awslabs/smithy-rs/issues/1114

## Description
Implement `From` trait for `SigningService`.

## Testing
Code change is minimal. Testing not added, changes checked for compilation and functionally through [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d8d6a2448d3500b6b45c51319fb048ef) as well as `cargo test`.

## Checklist
[x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.next.toml [hunk 1]
```diff
@@ -1,24 +1,5 @@
...
-[[aws-sdk-rust]]
-message = "Convert several `info` spans to `debug` in aws-config"
-references = ["smithy-rs#1087"]
-meta = { "breaking" = false, "tada" = false, "bug" = false }
-author = "rcoh"
-
 [[smithy-rs]]
-message = "Improve docs on `Endpoint::{mutable, immutable}`"
-references = ["smithy-rs#1087"]
+message = "Provide SigningService creation via owned String"
```
Comment (jdisanti) [inline CHANGELOG.next.toml:2]: In the changelog, keep the entries that were here already and just add your new entry.

File: aws/rust-runtime/aws-types/src/lib.rs [hunk 2]
```diff
@@ -42,3 +42,9 @@ impl SigningService {
         SigningService(Cow::Borrowed(service))
     }
 }
+
+impl From<String> for SigningService {
```
Comment (jdisanti) [inline aws/rust-runtime/aws-types/src/lib.rs:46]: This looks great! Thanks!

File: CHANGELOG.next.toml (hunk 1 above)
Comment (mchoicpe-amazon) [reply CHANGELOG.next.toml:2]: Shoot. Will fix that, thanks.

Comment (mchoicpe-amazon) [reply CHANGELOG.next.toml:27]: Hopefully new push was correctly picked up. Marking as resolved.

Comment (jdisanti) [reply CHANGELOG.next.toml:27]: Looks like some new entries snuck in during the time in between 😅 Needs another fix

**Review by jdisanti — APPROVED** [review]
Thank you for the contribution! ❤️

**Review by rcoh — APPROVED** [review]

**Comment by jdisanti** [comment]
Don't worry about the PR Bot failures. If everything else passes, I'll get this merged and into the next release.

**Pull request merged**